services/settings-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/SettingsServiceDAL.java

```diff
@@ -330,6 +334,18 @@ private Parameter putParameter(Parameter parameter) {
         return updated;
     }
 
+    private void deleteParameter(Parameter parameter) {
+        try {
+            DeleteParameterResponse response = ssm.deleteParameter(request -> request
+                    .name(parameter.name())
+            );
+        } catch (SdkServiceException ssmError) {
+            LOGGER.error("ssm:DeleteParameter error " + ssmError.getMessage());
+            throw ssmError;
+        }
```

Comment:
This could be a [ParameterNotFoundException](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/ssm/model/ParameterNotFoundException.html). Does it make sense to throw the error upwards in this case? Trying to delete a Parameter but not finding it means it doesn't exist, so the operation should be successful, right?

Reply:
Good design question. Do you want to know that you're trying to delete something that doesn't exist, or should that just be a warning (and swallow the `ParameterNotFoundException`)? I'm ok either way.